### DISCUSSION TITLE
[Feature] Redetect Faces

### DISCUSSION THREAD

**Discussion by dafeist-github** [immich-app/immich#7440] [Feature Request]
### The feature

A feature adding a button in the admin-page to let the server try to detect new people on ALL photos no matter if people there are already assigned. For example, the model got updated or the AI learned more: Can detect more faces later

### Platform

- [X] Server
- [x] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Is this not covered by the existing options in the jobs panel?

**Reply by dafeist-github** [reply]
No, if you try to rerun Face-Detection all your faces get cleared. I want the AI to detect more faces on the photos and add it to the existing database. This is helpful if you for example change some settings in detection or the AI model gets better etc.

**Reply by bo0tzz** [reply]
@mertalev thoughts? Is this feasible?

**Reply by mertalev** [reply]
It's possible, but adds complexity. If you change the detection score and/or model, there's no guarantee the face bounding boxes will be the same as before for that asset. So we'd have to get into the territory of merging bounding boxes and setting a similarity threshold for that. I don't like the idea of adding more moving parts to the facial recognition pipeline if we can avoid it.

But what's more common is that the face has been detected, but doesn't show up in the UI since it hasn't been assigned to a person. All that might be needed is a UI change to show those faces and allow users to assign them to people.

**Reply by dafeist-github** [reply]
Can I somehow see if a person on an image is detected as a single person?
